beaker-vue/src/components/BeakerHeader.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+    <Toolbar class="toolbar">
+        <template #start>
+            <div class="status-bar">
+                <i class="pi pi-circle-fill" :style="`font-size: inherit; color: var(--${connectionColor});`" />
+                {{capitalize(props.connectionStatus)}}
+            </div>
+            <Button
+                outlined
+                size="small"
+                icon="pi pi-angle-down"
+                iconPos="right"
+                class="connection-button"
+                @click="selectKernel"
+                :label="props.kernel"
+                :loading="props.loading"
+            />
+        </template>
+
+        <template #center>
+            <div class="logo">
+                <h4>
+                    Beaker <span class="longer-title">Development Interface</span>
+                </h4>
+            </div>
+        </template>
+
+        <template #end>
+            <nav>
+                <Button
+                    text
+                    @click="props.toggleDarkMode"
+                    style="margin: 0; color: var(--gray-500);"
+                    :icon="themeIcon"
+                />
+                <a
+                    href="https://jataware.github.io/beaker-kernel"
+                    rel="noopener"
+                    target="_blank"
+                >
+                    <Button
+                        text
+                        style="margin: 0; color: var(--gray-500);"
+                        aria-label="Beaker Documentation"
+                        icon="pi pi-book"
+                    />
+                </a>
+                <a
+                    href="https://github.com/jataware/beaker-kernel"
+                    rel="noopener"
+                    target="_blank"
+                >
+                    <Button
+                        text
+                        style="margin: 0; color: var(--gray-500);"
+                        aria-label="Github Repository Link Icon"
+                        icon="pi pi-github"
+                    />
+                </a>
+            </nav>
+        </template>
+    </Toolbar>
+</template>
+
+<script setup>
+import { defineProps, defineEmits, computed } from "vue";
+import Toolbar from 'primevue/toolbar';
+import Button from 'primevue/button';
+import { capitalize } from '../util';
+
+// TODO too many granular props- use a slot instead?
+const props = defineProps([
+    "connectionStatus",
+    "toggleDarkMode",
+    "isDarkMode",
+    "loading",
+    "kernel"
+]);
+
+const emit = defineEmits(["selectKernel"]);
+
+function selectKernel() {
+    emit('select-kernel');
+}
+
+const themeIcon = computed(() => {
+    return `pi pi-${props.isDarkMode ? 'sun' : 'moon'}`;
+});
+
+
+// TODO export/import from ts-lib utils.ts
```

Comment:
Having problems with typescript types on new files and/or files outside of `components/`.